kyaml/setters2/list.go

```diff
@@ -18,6 +18,8 @@ type List struct {
 	Name string
 
 	Setters []SetterDefinition
+
+	Substitutions []SubstitutionDefinition
```

Comment:
This feels like two separate pieces of functionality that maybe should be separate. Also, the implementation for setters and substitutions look very similar. Could we improve this?

Reply:
Setters and substitutions are closely coupled as substitution is combination of existing setters. So listing them together would make it easy for users to deduce relation between them.
For implementation, yes they seem similar in terms of parsing openAPI file, but the metadata is different so better to keep them separate. Updated the method signatures for distinction.